src/elements/content-sidebar/activity-feed/task-new/Task.js

```diff
@@ -54,122 +65,186 @@ const getMessageForTask = (isCurrentUser: boolean, taskType: TaskType) => {
-        // can_edit: permissions.can_update,
-        can_edit: false,
+class Task extends React.Component<Props, State> {
+    state = {
+        modalError: null,
+        isEditing: false,
+    };
+
+    handleEditClick = (): void => {
+        this.setState({ isEditing: true });
+    };
+
+    handleModalClose = () => {
+        this.setState({ isEditing: false });
     };
 
-    const currentUserAssignment =
-        assigned_to && assigned_to.entries
-            ? assigned_to.entries.find(({ target }) => target.id === currentUser.id)
-            : null;
-
-    const shouldShowActions =
-        currentUserAssignment &&
-        currentUserAssignment.permissions &&
-        currentUserAssignment.permissions.can_update &&
-        currentUserAssignment.status === TASK_NEW_NOT_STARTED &&
-        (status === TASK_NEW_NOT_STARTED || status === TASK_NEW_IN_PROGRESS);
-
-    const TaskTypeIcon = task_type === TASK_TYPE_APPROVAL ? IconTaskApproval : IconTaskGeneral;
-
-    return (
-        <div className="bcs-task-container">
-            {error ? <CommentInlineError {...error} /> : null}
-            <div
-                className={classNames('bcs-task', {
-                    'bcs-is-pending': isPending || error,
-                })}
-                data-testid="task-card"
-            >
-                <Comment
-                    avatarRenderer={avatar => (
-                        <div className="bcs-task-avatar">
-                            {avatar}
-                            <TaskTypeIcon width={20} height={20} className="bcs-task-avatar-badge" />
-                        </div>
-                    )}
-                    created_at={created_at}
-                    created_by={created_by.target}
-                    currentUser={currentUser}
-                    id={id}
-                    isPending={isPending}
-                    onDelete={onDelete}
-                    onEdit={onEdit}
-                    permissions={taskPermissions}
-                    tagged_message={description}
-                    translatedTaggedMessage={translatedTaggedMessage}
-                    translations={translations}
-                    type={COMMENT_TYPE_TASK}
-                    getAvatarUrl={getAvatarUrl}
-                    getUserProfileUrl={getUserProfileUrl}
-                    mentionSelectorContacts={mentionSelectorContacts}
-                    getMentionWithQuery={getMentionWithQuery}
-                    userHeadlineRenderer={userLinkInstance => {
-                        return (
-                            <FormattedMessage
-                                {...getMessageForTask(!!currentUserAssignment, task_type)}
-                                values={{ user: userLinkInstance }}
+    handleCreateSuccess = () => {
+        this.setState({ isEditing: false });
+    };
+
+    handleCreateError = (e: ElementsXhrError) => {
+        this.setState({ modalError: e });
+    };
+
+    getUsersFromTask = (): SelectorItems => {
```

Comment:
This takes the existing data in `assigned_to` (of type `TaskAssigneeCollection`) to something consumable by `PillSelector`. The output should conform to the flow type for `SelectorItems`

(This will be refined once when integrating read/write capability)